pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -32,44 +51,69 @@ async def _scan_packages():
     """
     scanned_packages = {}
     async with aiohttp.ClientSession() as session:
-        for content in iter(content_queue.get, None):
-            package = _build_osv_data(content)
-            osv_data = json.dumps(package["osv_data"])
-            async with session.post(url=OSV_QUERY_URL, data=osv_data) as response:
+        for osv_data in iter(content_queue.get, None):
+            data = json.dumps(osv_data)
+            async with session.post(url=OSV_QUERY_URL, data=data) as response:
                 response_body = await response.text()
                 json_body = json.loads(response_body)
                 if json_body.get("vulns"):
                     package_name = "{package}-{version}".format(
-                        package=package["osv_data"]["package"]["name"],
-                        version=package["osv_data"]["version"],
+                        package=osv_data["package"]["name"],
+                        version=osv_data["version"],
                     )
                     scanned_packages[package_name] = json_body["vulns"]
     await sync_to_async(VulnerabilityReport.objects.get_or_create)(
-        vulns=scanned_packages,
+        vulns=scanned_packages, pulp_domain=get_domain()
     )
 
 
-def _get_content_from_repo(repo_version_pk: str):
+def _get_content_from_repo_version(repo_version_pk: str):
     """
     Populate content_queue Queue with the content_units found in RepositoryVersion
     """
     repo_version = RepositoryVersion.objects.get(pk=repo_version_pk)
     for content_unit in repo_version.content:
-        content_queue.put(content_unit.cast())
+        content = content_unit.cast()
+        ecosystem = _identify_package_ecosystem(content)
+        osv_data = _build_osv_data(content.name, ecosystem, content.version)
+        content_queue.put(osv_data)
+    content_queue.put(None)  # signal that there is no more content_units
+
+
+def _parse_npm_pkg_dependencies(package_lock_content):
+    """
+    Parse the package-lock.json file to extract the packages[name][dependencies] and
+    add them to content_queue Queue
+
+    notes:
+    - we are striping the "~" and "^" from versions because osv.dev has no support to version range
+    - the old/legacy packages[dependencies] field is not supported
+    """
+    temp_file = PulpTemporaryFile.objects.get(pk=package_lock_content, pulp_domain=get_domain())
```

Comment:
Same deal here with this `get`.
```suggestion
    temp_file = PulpTemporaryFile.objects.get(pk=package_lock_content)
```